### ISSUE TITLE
Missing automatic label in last subplot

### ISSUE THREAD

**Issue by adigitoleo** [GenericMappingTools/GMT.jl#938]
Julia 1.7.3
GMT 6.3
GMT.jl master

The last subplot is missing a label when `autolabel = true`:

```julia
gmtbegin()
subplot(grid = (2, 2), size = (14, 10), autolabel = true)
plot(rand(10), panel = 1)
plot(1:10, panel = 2)
plot(10:-1:1, panel = 3)
plot(rand(10), panel = 4)
gmtend(show = true)
```

Throws a warning: `end [WARNING]: subplot was never completed - plot items in last panel may be missing`.

Output:
![GMTplot](https://user-images.githubusercontent.com/34595875/177174605-6741d71b-aa0c-4620-8f0e-abdb4733ca75.png)

GMT command line doesn't seem to cause this, this script works as expected:

```
#!/bin/sh
gmt begin subplot png
  gmt subplot begin 2x2 -Baf -BWSen -F10/10 -A
  gmt subplot set
  gmt basemap -Rg -Baf
  gmt subplot set 0
  echo 100 0|gmt plot -Rg -Baf -Sc2c
  gmt subplot set
  gmt basemap -Rg -Baf
  gmt subplot set
  gmt basemap -Rg -Baf
  gmt subplot set
  gmt basemap -Rg -Baf
  gmt subplot end
gmt end
```

**Comment by joa-quim** [comment]
Same thing as in #936. But with the fix in #939 it now barfs with
```
Warning: ERROR USING SUBPLOT. You should not call gmtend() before subplot(:end)
└ @ GMT C:\Users\joaqu\.julia\dev\GMT\src\gmtbegin.jl:33
```
but produces the right plot.

**Comment by adigitoleo** [comment]
Solved on master
